src/public/app/containers/Song.js

```diff
@@ -177,119 +171,148 @@ class Song extends React.Component {
       credentials: 'include',
       body: JSON.stringify(track),
     })
-    .then(res => res.json())
-    .then(favs => this.props.handleFavoritesChange(favs))
-    .catch(err => console.log(err));
+      .then(res => res.json())
+      .then(favs => this.props.handleFavoritesChange(favs))
+      .catch(err => console.log(err));
   }
 
   handleFavoritesClick(track) {
     if (this.props.favorites.some(fav => fav.track_id === track.track_id)) {
       this.removeFavorite(track);
-    } else this.addFavorite(track);
+    } else {
+      this.addFavorite(track);
+    }
   }
 
-  addToQueue() { this.props.addTrackToSpotifyQueue(this.props.track); }
+  addToQueue() {
+    this.props.addTrackToSpotifyQueue(this.props.track);
+  }
 
   render() {
-    this.netSize = this.props.size - this.borderWidth;
+    const {
+      auth,
+      currentSong,
+      favorites,
+      handleExpandClick,
+      playlist,
+      ranking,
+      spotifyPlayer,
+      track_preview_url,
+      track,
+    } = this.props;
+  
     // TODO fix this piece of logic
```

Comment:
How about something like:

```
const isPlaying = spotifyPlayer.isPaused || currentSong.isPlaying
const isCurrentTrack = auth
  ? spotifyPlayer.currentTrack && spotifyPlayer.currentTrack.track_id === track.track_id
  : currentSong.src === track.track_preview_url

const icon = (isPlaying && isCurrentTrack) ? 'pause' : 'play'
```

would be nice if `isPlaying` and `currentTrack` were the same for both flows, but I think the above should do the trick, and should make the code a bit clearer.

Reply:
May need to change that to
```
const isPlaying = (spotifyPlayer.currentTrack && !spotifyPlayer.isPaused) || currentSong.isPlaying
```

looks like spotifyPlayer.isPaused defaults to `undefined`, so `!spotifyPlayer.isPaused` will be true initially, but checking for a current track and !pause should give us an accurate `isPlaying`

Reply:
Yeah I agree